List/List/List.c

```diff
@@ -0,0 +1,191 @@
+#include "List.h"
+#include <stdio.h>
+
+// A structure containing a pointer to the head of the list
+typedef struct List
+{
+    struct ListElement* head;
+} List;
+
+// A structure containing a pointer to the next list item and a value variable for the list items
+typedef struct ListElement
+{
+    int value;
+    struct ListElement* next;
+} ListElement;
+
+List* createList()
+{
+    return calloc(1, sizeof(List));
+}
+
+void deleteList(List* list)
+{
+    ListElement* element = list->head;
+    while (list->head != NULL)
+    {
+        list->head = list->head->next;
+        free(element);
+        element = list->head;
+    }
+    free(list);
+}
+
+void removeElement(ListElement* element, List* list, int* error)
+{
+    *error = 0;
+    if (element == NULL || element->next == NULL)
+    {
+        *error = 1;
+        return;
+    }
+    ListElement* temporary = element->next;
+    element->next = element->next->next;
+    free(temporary);
+}
+
+void removeFirstElement(List* list, int* error)
+{
+    *error = 0;
+    if (list->head == NULL)
+    {
+        *error = 2;
+        return;
+    }
+    ListElement* temporary = list->head;
+    list->head = list->head->next;
+    free(temporary);
+}
+
+ListElement* first(List* list)
+{
+    ListElement* element = list->head;
+    return element;
+}
+
+ListElement* next(ListElement* element)
+{
+    element = element->next;
+    return element;
```

Comment:
```suggestion
    return element->next;
```
Можно присваивать в element что угодно, это всё равно локальный параметр и никто этих изменений не увидит